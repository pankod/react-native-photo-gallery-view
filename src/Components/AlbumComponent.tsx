// Global Imports
import React, { Component } from 'react';
import { Text, View } from 'react-native';

// Local Imports
import { FooterComponent } from "@Components";
import { AlbumStyle } from "@Styles";
import { IAlbumProps } from "@Interfaces";

export class AlbumComponent extends Component<IAlbumProps, {}> {
	public render(): JSX.Element {
		const {
			customFooterStyle
		} = this.props;
		return (
			<React.Fragment>
				<View style={AlbumStyle.container}>
					<Text>Main</Text>
				</View>
				<FooterComponent customFooterStyle={{ ...customFooterStyle }} />
			</React.Fragment>
		)
	}
}
