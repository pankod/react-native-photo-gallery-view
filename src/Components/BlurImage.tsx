// Global Imports
import React, { PureComponent } from 'react'
import { Animated } from 'react-native'

// Local Imports
import { IBlurImageProps } from '@Interfaces';
import { BlurImageStyle } from '@Styles';

export class BlurImage extends PureComponent<IBlurImageProps> {

	public imageAnimated = new Animated.Value(0);

	private onImageLoad(): void {
		Animated.timing(this.imageAnimated, {
			toValue: 1,
			duration: 500,
			useNativeDriver: true
		}).start();
	}

	render() {
		return (
			<React.Fragment>
				<Animated.Image
					{...this.props}
					style={[BlurImageStyle.image, { opacity: this.imageAnimated }]}
					onLoad={this.onImageLoad.bind(this)}
					blurRadius={1}
				/>
			</React.Fragment>
		)
	}
}
