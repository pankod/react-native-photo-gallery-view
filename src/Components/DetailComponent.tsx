// Global Imports
import React, { PureComponent } from 'react'

// Local Imports
import Common from '@Provider';
import { IGalleryProps, IGalleryState } from '@Interfaces';
import { BlurImage } from '@Components';

export class DetailComponent extends PureComponent {

	static contextType = Common;

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{
					(context: IGalleryProps & IGalleryState) => (
						<BlurImage
							resizeMethod={"resize"}
							resizeMode={"contain"}
							source={{ uri: context.showingImage && context.showingImage.photo }}
							thumbnail={{ uri: context.showingImage && context.showingImage.thumb }}
						/>
					)
				}
			</Common.Consumer>
		)
	}
}
