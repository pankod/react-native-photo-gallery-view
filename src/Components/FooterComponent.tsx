// Global Imports
import React, { Component } from 'react';
import { Text, View } from 'react-native';

// Local Imports
import { FooterStyle } from "@Styles";
import { IFooterProps } from "@Interfaces";

export class FooterComponent extends Component<IFooterProps, {}> {
	public render(): JSX.Element {
		const { customFooterStyle } = this.props
		return (
			<View style={[FooterStyle.container, customFooterStyle]}>
				<Text> Footer Component </Text>
			</View>
		)
	}
}
