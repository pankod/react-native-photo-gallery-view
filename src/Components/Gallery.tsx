import React, { Component } from "react"
import { Text, View } from "react-native"

export class Gallery extends Component {
	render() {
		return (
			<View>
				<Text> textInComponent </Text>
			</View>
		)
	}
}
