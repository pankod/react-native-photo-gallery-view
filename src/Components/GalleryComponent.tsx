// Global Imports
import React, { Component } from "react"
import { View } from "react-native"

// Local Imports
import { TopBarComponent, AlbumComponent } from "@Components";
import { IGalleryProps, IGalleryState } from "@Interfaces";
import { GalleryStyle } from "@Styles";

export class GalleryComponent extends Component<IGalleryProps, IGalleryState> {
	public render(): JSX.Element {
		const {
			customFooterStyle,
			customTopBarStyle
		} = this.props;
		return (
			<View style={GalleryStyle.container}>
				<TopBarComponent customTopBarStyle={{ ...customTopBarStyle }} />
				<AlbumComponent
					customFooterStyle={{ ...customFooterStyle }}
				/>
			</View>
		)
	}
}
