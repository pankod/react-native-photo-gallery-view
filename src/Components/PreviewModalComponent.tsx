// Global Imports
import React, { PureComponent } from 'react'
import { Text, View, Modal, Image, Animated, ImageBackground, Platform } from 'react-native'

// Local Imports
import Common from '@Provider';
import { IGalleryProps, IGalleryState } from '@Interfaces';
import { PreviewModalStyle } from '@Styles';

export class PreviewModal extends PureComponent {

	static contextType = Common;

	public scale = new Animated.Value(0);
	public opacity = new Animated.Value(0);

	public componentDidMount(): void {
		Animated.parallel([
			Animated.spring(this.scale, {
				toValue: 1,
				bounciness: 10,
				useNativeDriver: true
			}),
			Animated.timing(this.opacity, {
				toValue: 1,
				duration: 500,
				useNativeDriver: true
			})
		]).start();
	}

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{(context: IGalleryProps & IGalleryState) => (
					<Modal
						animationType={"none"}
						visible={context.previewIsOpen}
						transparent={true}
					>
						<ImageBackground style={{
							flex: 1,
						}}
							blurRadius={Platform.OS == 'ios' ? 20 : 15}
							// source={{ uri: context.imagePreview.thumb }}
							source={require("../Assets/Images/radius-bg.png")}
							resizeMode={"cover"}
						>
							<Animated.View
								style={[
									PreviewModalStyle.container,
									{
										transform: [
											{
												scale: this.scale.interpolate({
													inputRange: [0, 1],
													outputRange: [0, 1]
												})
											}
										],
										opacity: this.opacity
									}
								]}>
								<View style={PreviewModalStyle.imagePreview}>
									{
										context.customPreviewComponent ?
											context.customPreviewComponent(context.imagePreview) :
											(
												<Image
													style={{ width: "100%", flex: 1 }}
													source={{ uri: context.imagePreview.photo }}
													resizeMode={"contain"}
												/>
											)
									}
								</View>
							</Animated.View>
						</ImageBackground>
					</Modal>
				)}
			</Common.Consumer>
		)
	}
}
