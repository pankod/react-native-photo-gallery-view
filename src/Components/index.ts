export { GalleryComponent } from '@Components/GalleryComponent';
export { TopBarComponent } from '@Components/TopBarComponent';
export { FooterComponent } from '@Components/FooterComponent';
export { AlbumComponent } from '@Components/AlbumComponent';
