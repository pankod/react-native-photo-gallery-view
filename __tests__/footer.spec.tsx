
import * as React from 'react';
import { View } from 'react-native';

import renderer from "react-test-renderer";
import { ShallowWrapper, shallow, mount } from "enzyme";

import { FooterComponent } from "../src/Components/FooterComponent";

describe("Footer Component", () => {
	let wrapper: ShallowWrapper;
	let mounting;
	const props = {

	}
	let component = (
		<FooterComponent {...props} />
	);

	beforeEach(() => {
		wrapper = shallow(component);
		mounting = mount(component);
	});

	test('should Footer Component is render correctly', () => {
		const render = renderer.create(component).toJSON();

		expect(render).toBeDefined();
		expect(render).toMatchSnapshot();
	});
});
