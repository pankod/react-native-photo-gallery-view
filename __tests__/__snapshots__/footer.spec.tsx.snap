// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should Footer Component is render correctly 1`] = `
<react-native-mock>
  <react-native-mock>
     Footer Component 
  </react-native-mock>
</react-native-mock>
`;
